install/installation/installation.go

```diff
@@ -397,23 +416,6 @@ func (k KymaInstaller) applyInstallationCR(installationCR *v1alpha1.Installation
 	return nil
 }
 
-// extractInstallationCR finds and removes first Installation CR from the slice of K8sObjects and returns it
-func (k KymaInstaller) extractInstallationCR(k8sObjects []k8s.K8sObject) (*v1alpha1.Installation, []k8s.K8sObject, error) {
-	for i, k8sObject := range k8sObjects {
-		if k8sObject.GVK.Group == v1alpha1.Group && k8sObject.GVK.Kind == "Installation" {
-			installationCR, ok := k8sObject.Object.(*v1alpha1.Installation)
-			if !ok {
-				return nil, nil, fmt.Errorf("unexpected type of Installation object: %T, failed to cast to *Installation", k8sObject.Object)
-			}
-
-			k8sObjects = append(k8sObjects[:i], k8sObjects[i+1:]...)
-			return installationCR, k8sObjects, nil
-		}
-	}
-
-	return nil, k8sObjects, fmt.Errorf("installation object not found in the objects slice")
-}
-
```

Comment:
To properly handle artifacts with and without the Installation CR Yaml we could add an error that would provide distinction between when something went wrong and when Installation CR is not present if Yaml file.
We can do it as follows:
```go
var (
	errorInstallationCRNotFound = fmt.Errorf("error installationCR not found in objects")
)

// extractInstallationCR finds and removes first Installation CR from the slice of K8sObjects and returns it
func (k KymaInstaller) extractInstallationCR(k8sObjects []k8s.K8sObject) (*v1alpha1.Installation, []k8s.K8sObject, error) {
	for i, k8sObject := range k8sObjects {
		if k8sObject.GVK.Group == v1alpha1.Group && k8sObject.GVK.Kind == "Installation" {
			installationCR, ok := k8sObject.Object.(*v1alpha1.Installation)
			if !ok {
				return nil, nil, fmt.Errorf("unexpected type of Installation object: %T, failed to cast to *Installation", k8sObject.Object)
			}

			k8sObjects = append(k8sObjects[:i], k8sObjects[i+1:]...)
			return installationCR, k8sObjects, nil
		}
	}

	return nil, k8sObjects, errorInstallationCRNotFound
}
```
or use some additional bool variable.